### ISSUE TITLE
Issue parsing/formatting GB "special numbers"

### ISSUE THREAD

**Issue by dieselmachine** [daviddrysdale/python-phonenumbers#172]
The following numbers, when parsed via https://libphonenumber.appspot.com/ (using GB as the default country code), all yield the same result.

8001111
08001111
+448001111
+4408001111

the result is:

E164             +448001111
INTERNATIONAL    +44 800 1111
NATIONAL         0800 1111

the python library, up until 8.7.1, returned the same result. Starting with 8.8.0, a leading zero, if present, is reproduced in the NATIONAL format, yielding 2 different results. If the leading zero is omitted, the results match the expected ones above. If a leading zero is present, I get the following result:

E164             +4408001111 (should not be a zero in here)
INTERNATIONAL    +44 08001111 (should not be a zero in here)
NATIONAL         08001111

It looks like the issue is the length check, which now fails to strip the leading zero if the length check returns ValidationResult.IS_POSSIBLE_LOCAL_ONLY.

**Comment by daviddrysdale** [comment]
Thanks for reporting this, and thanks in particular for narrowing down the point at which results started diverging from
upstream.

On an initial look at internals, it seems that the change in v8.8.0 actually just made visible a divergence that was already present: the Java version of the code has `PhoneNumberDesc.possibleLengthLocalOnly_` with values `[4, 5, 6, 8]` for `GB` numbers
of an `UNKNOWN` type, whereas the Python version has:
```Python
    general_desc=PhoneNumberDesc(national_number_pattern='[1-357-9]\\d{9}|[18]\\d{8}|8\\d{6}', possible_length=(7, 9, 10), possible_length_local_only=(4, 5, 6, 7, 8)),
```

(which includes `7` as a `possible_length_local_only` value, unlike Java).

This means that the Python metadata conversion/generation process appears to be at fault, possibly since 5599c56165a30 (related to the changes [described here](https://groups.google.com/g/libphonenumber-discuss/c/75TOpTFVi08)).

 I think the relevant chunk of upstream metadata generation code is [here](https://github.com/google/libphonenumber/blob/master/tools/java/common/src/com/google/i18n/phonenumbers/BuildMetadataFromXml.java#L681-L731), in particular the part that avoids adding a length value to the general `possible_length_local_only` if that value is already a `possible_length`.

Initial stab at a fix in #173, but I want to think about it a bit more before applying&hellip;

**Comment by dieselmachine** [comment]
That initial stab definitely fixes the issue i was experiencing, for the record. Everything is formatting beautifully now. Thanks!
